ilastik/applets/layerViewer/layerViewerGui.py

```diff
@@ -310,6 +356,72 @@ def _create_grayscale_layer_from_slot(cls, slot, n_channels):
         return layer
 
     @classmethod
+    def createGrayscaleLayer(cls, slot):
```

Comment:
The function createGrayscaleLayer seems to be redundant as you said. I found just the method GrayscaleLayer and this needs a source/lazyflowsource. Maybe you add a comment on how you use the addGrayscaleLayer and replace the one I used in the code. 
So that next programmers easily find the function and know how to use it. 

I guess, _create_single_color_layer_from_slot and create_8bit_ordered_random_colortable_layer_from_slot are not needed either. but the one with transparent zero is used.